src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -89,7 +89,11 @@ private void OnDataReceived(object sender, DataReceivedEventArgs e)
         {
             // This does not need to be inside of a critical section.
             // The logging queues and command handlers are thread-safe.
-            ExecutionContext.Output(e.Data);
+            var commandHandler = HostContext.GetService<ICommandHandler>();
```

Comment:
I would move GetService<ICommandHandler> to the initialize method on the handler base class and store it in a private property.